src/compute.geometry/GrasshopperDefinition.cs

```diff
@@ -409,22 +411,17 @@ public Schema Solve()
             Schema outputSchema = new Schema();
             outputSchema.Algo = "";
 
-            foreach(var kvp in _output)
+            // solve definition
+            Definition.Enabled = true;
+            Definition.NewSolution(true, GH_SolutionMode.CommandLine);
```

Comment:
This might be really inefficient. We may want to make it an option.

It would help me to better understand what this is fixing

Reply:
In #118 we discussed having Compute solve Grasshopper definitions in a way that more closely matched what happens in Grasshopper itself, and in the GrasshopperPlayer. In particular, this solves issues with solve order when there are multiple disjoint graphs on the canvas. The ability to only solve part of the graph to populate a specific output is nice, but from talking to people I feel this is more a cause of confusion than a feature, particularly when it comes to the order in which outputs are solved.

Or are you referring to the use of `expireAllObjects = true` when calling `NewSolution()`?

Reply:
I would rather we ensure we solve the entire solution the first time it is loaded and then perform partial solves.

Reply:
Changed in dcd70e4 to solve without expiring all objects. This means that if we were to do an initial solve before caching the definition, it would only solve the necessary parts.